service/controller/resource/tccp/create.go

```diff
@@ -319,11 +319,16 @@ func (r *Resource) newParamsMain(ctx context.Context, cr infrastructurev1alpha2.
 }
 
 func (r *Resource) newParamsMainInstance(ctx context.Context, cr infrastructurev1alpha2.AWSCluster, t time.Time) (*template.ParamsMainInstance, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
 	var instance *template.ParamsMainInstance
 	{
 		instance = &template.ParamsMainInstance{
 			Master: template.ParamsMainInstanceMaster{
-				AZ: key.MasterAvailabilityZone(cr),
+				AZ: cc.Spec.TenantCluster.TCCP.AvailabilityZones[0].Name,
```

Comment:
Isn't this code we need to drop? What is this doing? Magically taking the first AZ here does also not seem to be right.

Reply:
The functionality stay the same, the function did basically the same thing anyway.
The difference is that I wanted to get rid of the `key.MasterAvailabilityZone(cr),` function which took info from the Cluster resource which is not right ( and we will be removing the master node info from the Cluster anyway).

Currently, this part is there to have an empty ETCD volume on the right availability zone to do a snapshot and then create volume from that snapshot in the tccpn stack. This is part of the migration code we added when introducing the new stack.

We can take a look at how we could remove it before creating release but I don't want to invest too much time as it could take some serious time and i would rather try to work on important parts.

Reply:
Can you put a comment describing that so that we do not wonder too much about it? Just copy/paste some parts of what you just wrote here and put it in the code.